lib/charms/haproxy/v0/haproxy_route.py

```diff
@@ -0,0 +1,1191 @@
+        description="Which rewrite method to apply.One of set-path, set-query, set-header."
+    )
+    expression: str = Field(description="Regular expression to use with the rewrite method.")
+    header: Optional[str] = Field(description="The name of the header to rewrite.", default=None)
+
+
+class RequirerApplicationData(_DatabagModel):
+    """Configuration model for HAProxy route requirer application data.
+
+    Attributes:
+        service: Name of the service requesting HAProxy routing.
+        ports: List of port numbers on which the service is listening.
+        paths: List of URL paths to route to this service. Defaults to an empty list.
+        subdomains: List of subdomains to route to this service. Defaults to an empty list.
+        rewrites: List of RewriteConfiguration objects defining path, query, or header
+            rewrite rules.
+        check: ServerHealthCheck configuration for monitoring backend health.
+        load_balancing: Configuration for the load balancing strategy.
+        rate_limit: Optional configuration for limiting connection rates.
+        bandwidth_limit: Optional configuration for limiting upload and download bandwidth.
+        retry: Optional configuration for request retry behavior.
+        deny_paths: List of URL paths that should not be routed to the backend.
+        timeout: Configuration for server, client, and queue timeouts.
+        server_maxconn: Optional maximum number of connections per server.
+    """
+
+    service: str = Field(description="The name of the service.")
+    ports: list[int] = Field(description="The list of ports listening for this service.")
+    paths: list[str] = Field(description="The list of paths to route to this service.", default=[])
+    subdomains: list[str] = Field(
+        description="The list of subdomains to route to this service.", default=[]
+    )
+    rewrites: list[RewriteConfiguration] = Field(
+        description="The list of path rewrite rules.", default=[]
+    )
+    check: ServerHealthCheck = Field(
+        description="Configure health check for the service.",
+        default=ServerHealthCheck(interval=60, rise=2, fall=3),
+    )
+    load_balancing: LoadBalancingConfiguration = LoadBalancingConfiguration(
+        algorithm=LoadBalancingAlgorithm.LEASTCONN
+    )
+    rate_limit: Optional[RateLimit] = Field(
+        description="Configure rate limit for the service.", default=None
+    )
+    bandwidth_limit: Optional[BandwidthLimit] = Field(
+        description="Configure bandwidth limit for the service.", default=None
+    )
+    retry: Optional[Retry] = Field(
+        description="Configure retry for incoming requests.", default=None
+    )
+    deny_paths: list[str] = Field(
+        description="Configure path that should not be routed to the backend", default=[]
+    )
+    timeout: TimeoutConfiguration = Field(
+        description="Configure timeout",
+        default=TimeoutConfiguration(server=60, client=60, queue=60),
```

Comment:
We should move the default value to the TimeoutConfiguration definition

Reply:
Moved those default values to model configuration